crates/common/src/commit/client.rs

```diff
@@ -0,0 +1,80 @@
+use alloy_rpc_types_beacon::{BlsPublicKey, BlsSignature};
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION};
+use serde::{Deserialize, Serialize};
+
+use super::{error::SignerClientError, request::SignRequest};
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct GetPubkeysResponse {
+    pub consensus: Vec<BlsPublicKey>,
+    pub proxy: Vec<BlsPublicKey>,
+}
+
+const GET_PUBKEYS_PATH: &str = "/signer/v1/pubkeys";
+const REQUEST_SIGNATURE_PATH: &str = "/signer/v1/request_signature";
```

Comment:
I'm not sure at all why the change of the API was necessary and why these new constants have been introduced.
Seeing that we don't have many users yet, I guess changing APIs is fine, but this generally comes with a lot of precaution for me :D

[NIT] In any case, I'd think about unifying the `SignerClient`'s method names with the endpoints. Depends on how much we wanna think about this at this stage, but it's good to keep them corresponding.

I'd change the `pubkeys` endpoint to `get_pubkeys`, so that we have both `get_pubkeys` and `request_signature` starting with a verb :smile:

Reply:
Oh, I see, that's what I was missing as context, it's clear now. A second PR is fine 👍